lib/chcol/json.go

```diff
@@ -25,6 +25,38 @@ import (
 	"strings"
 )
 
+// JSONSerializer interface allows a struct to be manually converted to an optimized JSON structure instead of relying
+// on recursive reflection.
+// Note that the struct must be a pointer in order for the interface to be matched, reflection will be used otherwise.
+type JSONSerializer interface {
+	ToClickHouseJSON() (*JSON, error)
```

Comment:
```suggestion
	SerializeClickHouseJSON() (*JSON, error)
```

nit: I wonder if it should be named after interface

Reply:
Good idea. I tried to make these names as distinct as possible so that they wouldn't conflict with any other interfaces such as `SerializeJSON` or `MarshalJSON`. It's a bit long but it is clearer